pkg/controllers/namespace/namespaceController.go

```diff
@@ -0,0 +1,285 @@
+type NamespaceController struct {
+	indexer        cache.Indexer
+	informer       cache.Controller
+	calicoObjCache calicocache.ResourceCache
+	calicoClient   *client.Client
+}
+
+// NewNamespaceController Constructor for NamespaceController
+func NewNamespaceController(k8sConfig *rest.Config) controller.Controller {
+
+	// Get kubenetes clientset
+	clientset, err := kubernetes.NewForConfig(k8sConfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	cconfig, err := client.LoadClientConfig("")
+	if err != nil {
+		panic(err)
+	}
+
+	// Get Calico client
+	cclient, err := client.New(*cconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// Function returns slice of profile objects stored by policy controller
+	// on ETCD datastore.Indentifies controller writen objects by
+	// their naming convention.
+	listFunc := func() ([]interface{}, error) {
+
+		var filteredProfiles []interface{}
+
+		// Get all profile objects from ETCD datastore
+		calicoProfiles, err := cclient.Profiles().List(api.ProfileMetadata{})
+		if err != nil {
+			return filteredProfiles, err
+		}
+
+		// Filter out only objects that are written by policy controller
+		for _, profile := range calicoProfiles.Items {
+
+			profileName := profile.Metadata.Name
+			if strings.HasPrefix(profileName, converter.ProfileNameFormat) {
+				filteredProfiles = append(filteredProfiles, profile)
+			}
+		}
+		glog.Debugf("Found %d profiles in calico ETCD:", len(filteredProfiles))
+		return filteredProfiles, nil
+	}
+
+	// Function returns key of the object in kubernetes format.
+	keyFunc := func(obj interface{}) string {
+		profile := obj.(api.Profile)
+		Key := profile.Metadata.Name
+		return Key
+	}
+
+	cacheArgs := calicocache.ResourceCacheArgs{
+		ListFunc:   listFunc,
+		KeyFunc:    keyFunc,
+		Client:     cclient,
+		ObjectType: reflect.TypeOf(&api.Profile{}), // Restrict cache to store pointers to calico profile objects only.
+	}
+	
+	ccache := calicocache.NewResourceCache(cacheArgs)
+	namespaceConverter := converter.NewNamespaceConverter()
+
+	// create the watcher
+	listWatcher := cache.NewListWatchFromClient(clientset.Core().RESTClient(), "namespaces", "", fields.Everything())
+
+	// Bind the calico cache to kubernetes cache with the help of an informer. This way we make sure that
+	// whenever the kubernetes cache is updated, changes get reflected in calico cache as well.
+	indexer, informer := cache.NewIndexerInformer(listWatcher, &v1.Namespace{}, 0, cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			glog.Infof("Got ADD event for namespace: %s\n", key)
+
+			if err != nil {
+				glog.Error(err)
+				return
+			}
+
+			profile, conversionErr := namespaceConverter.Convert(obj)
+			if conversionErr != nil {
+				glog.Errorf("Error while converting %#v to calico profile.", obj)
+				return
+			}
+			// Add profileName:*profile in calicoCache
+			ccache.Set(profile.(*api.Profile).Metadata.Name, profile)		
```

Comment:
Another case for using the `key` from above (Kubernetes key) - we could make all of this informer code common if we did so.

We'd just need to pass it a converter and it could do the rest.